Switch docs theme to furo

marksweb opened the pull request:
## Description

Following discussion in the documentation work group, this switches the docs theme to furo.

crydotsnake reviewed (changes requested):
We could change Divio AG too djangoCMS Association in the Copyright section

vinitkumar commented:
@marksweb There are tests failing as well due to missing dependencies. Please fix them too.

marksweb commented:
> @marksweb There are tests failing as well due to missing dependencies. Please fix them too.

That highlights a problem with those tests. The sphinx requirements are in the docs requirements, so the tests need reviewing that are failing as to what they're trying to do.

marksweb commented:
@vinitkumar So it turns out there's a unit test which tests the docs, so there were some of the docs requirements in the base test requirements.

No need for this duplication, so I've removed the docs requirements from the test requirements and added an install step for the docs requirements in the unit tests. Things are starting to pass now. I'll address @crydotsnake point next

crydotsnake commented on docs/conf.py:
```diff
@@ -64,7 +64,7 @@
 # General information about the project.
 project = 'django cms'
-copyright = '2009-{}, Divio AG and contributors'.format(current_year)
+copyright = '2009-{}, djangoCMS Association and contributors'.format(current_year)
```
Should we also change the year here? Because the djangoCMS Association exists since 2020 i think.

FinalAngel replied:
should be "django CMS Association" though

marksweb replied:
@FinalAngel yeah good point, updated.

vinitkumar reviewed (approved):
Looks good to me. Let's get this merged and released 🚢  🚀
